src/App.js

```diff
@@ -1,11 +1,56 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
 import './App.css';
-
+import Pokemon from "./assets/Pokemon";
+import Button from "./assets/Button";
+//TEST
 function App() {
 
+    const [pokemon, setPokemon] = useState(null);
+    const [url, setUrl] = useState('https://pokeapi.co/api/v2/pokemon');
+    const [loading, toggeleLoading] = useState(false);
+    const [error, setError] = useState(false);
+
+    useEffect(() => {
+
+        const token = axios.CancelToken.source();
+
+        async function fetchData() {
+            toggeleLoading(true);
+            setError(false);
+
+            try {
+                const result = await axios.get(url);
+                setPokemon(result.data);
+                toggeleLoading(false);
+
+            } catch (e) {
+                console.error(e);
+                toggeleLoading(false);
+                setError(true);
+            }
+        }
+        fetchData();
+
+        return () => {
+            token.cancel('Operation canceled by user!')
+        }
+
+    }, [url]);
 
     return (
-        <div>
+        <div className='pokemon-overview'>
+            { pokemon && <Button previousUrl={pokemon.previous} nextUrl={pokemon.next} setUrl={setUrl} />}
```

Comment:
Heel goed dat je een apart component hebt gebruikt voor de knoppen. Ik zou het component echter wel een naam geven die ook daadwerkelijk beschrijft wat het is, namelijk Button**s**